amlip_cpp/test/blackbox/communication/collaborative_learning/modelManagerTimeoutReplyTest.cpp

```diff
@@ -0,0 +1,167 @@
+
+    TestModelReplier()
+    {
+        // Do nothing
+    }
+
+    virtual eprosima::amlip::types::ModelReplyDataType fetch_model (
+            const eprosima::amlip::types::ModelRequestDataType data) override
+    {
+        logUser(AMLIPCPP_MANUAL_TEST, "Processing data: " << data << " . Processing data...");
+
+        eprosima::amlip::types::ModelReplyDataType solution;
+        if (data.to_string() == "MobileNet V1")
+        {
+            solution = eprosima::amlip::types::ModelReplyDataType("MOBILENET V1");
+        }
+        else if (data.to_string() == "MobileNet V2")
+        {
+            solution = eprosima::amlip::types::ModelReplyDataType("MOBILENET V2");
+        }
+        else
+        {
+            solution = eprosima::amlip::types::ModelReplyDataType("Do not have this model :,(");
+        }
+        logUser(AMLIPCPP_MANUAL_TEST, "Processed model: " << solution << " . Returning model...");
+
+        return solution;
+    }
+
+};
+
+} /* namespace test */
+
+using namespace eprosima::amlip;
+
+/**
+ * Launch 1 ModelManagerReceiverNode and 2 ModelManagerSenderNode.
+ *
+ * Models will be mocked with strings.
+ */
+TEST(modelManagerTimeoutReplyTest, ping_pong)
+{
+    // Activate log
+    eprosima::utils::Log::SetVerbosity(eprosima::utils::Log::Kind::Info);
+    {
+
+        // Managers always send same model in this test
+
+        // Create ModelManagerReceiver Node
+        eprosima::amlip::types::AmlipIdDataType id_receiver("ModelManagerReceiver");
+        // NOTE: this data must be created before nodes or nodes must stop before this is destroyed
+        eprosima::amlip::types::ModelRequestDataType data("MobileNet V1");
+        eprosima::amlip::node::ModelManagerReceiverNode model_receiver_node(id_receiver, data);
+
+        logUser(AMLIPCPP_MANUAL_TEST, "Node created: " << model_receiver_node << ". Creating model...");
+
+        // Create ModelManagerSender Nodes
+        eprosima::amlip::types::AmlipIdDataType id_sender_1("ModelManagerSender_1");
+        eprosima::amlip::node::ModelManagerSenderNode model_sender_node_1(id_sender_1);
+
+        eprosima::amlip::types::AmlipIdDataType id_sender_2("ModelManagerSender_2");
+        eprosima::amlip::node::ModelManagerSenderNode model_sender_node_2(id_sender_2);
+
+        // Create statistics data
+        std::string data_str_1 = "Hello world, I'm going to die.";
+        model_sender_node_1.publish_statistics("v0", data_str_1);
+
+        // Create waiter_receiver
+        std::shared_ptr<eprosima::utils::event::BooleanWaitHandler> waiter_receiver =
+                std::make_shared<eprosima::utils::event::BooleanWaitHandler>(false, true);
+
+        // Create listener
+        std::shared_ptr<test::TestModelListener> listener =
+                std::make_shared<test::TestModelListener>(waiter_receiver);
+
+        std::shared_ptr<test::TestModelReplier> replier =
+                std::make_shared<test::TestModelReplier>();
+
+        model_sender_node_1.stop();
```

Comment:
Model statistics are sent due to transient local qos event if the node is stopped. However, probably this should not happen. I agree it would be complicated to make this test in some other way (somehow aborting from the fetch_model callback), so at least please write a nice comment with the explanation.